statsmodels/tsa/statespace/_initialization.pxd

```diff
@@ -0,0 +1,111 @@
+#cython: boundscheck=False
+#cython: wraparound=False
+#cython: cdivision=False
+"""
+State Space Models - Initialization declarations
+
+Author: Chad Fulton  
+License: Simplified-BSD
+"""
+
+cimport numpy as np
+
+from statsmodels.tsa.statespace._representation cimport (
+    sStatespace, dStatespace, cStatespace, zStatespace
+)
+
+cdef class sInitialization(object):
+    cdef readonly int k_states
+    cdef public np.float64_t approximate_diffuse_variance
+
+    cdef np.float32_t [:] constant
+    cdef np.float32_t [::1, :] stationary_cov
+    cdef np.float32_t [::1, :] _tmp_transition
+    cdef np.float32_t [::1, :] _tmp_selected_state_cov
+
+    cpdef int initialize(self, inititialization_type, int offset,
+                         sStatespace model,
+                         np.float32_t [:] initial_state_mean,
+                         np.float32_t [::1, :] initial_diffuse_state_cov,
+                         np.float32_t [::1, :] initial_stationary_state_cov,
+                         int complex_step=*) except 1
+    cdef int clear_constant(self, int offset, np.float32_t [:] initial_state_mean) except 1
+    cdef int clear_cov(self, int offset, np.float32_t [::1, :] cov) except 1
+    cdef int initialize_known_constant(self, int offset, np.float32_t [:] initial_state_mean) except 1
+    cdef int initialize_known_stationary_cov(self, int offset, np.float32_t [::1, :] initial_stationary_state_cov) except 1
+    cdef int initialize_diffuse(self, int offset, np.float32_t [::1, :] initial_diffuse_state_cov) except 1
+    cdef int initialize_approximate_diffuse(self, int offset, np.float32_t [::1, :] initial_stationary_state_cov) except 1
+    cdef int initialize_stationary_constant(self, int offset, sStatespace model, np.float32_t [:] initial_state_mean, int complex_step=*) except 1
+    cdef int initialize_stationary_stationary_cov(self, int offset, sStatespace model, np.float32_t [::1, :] initial_stationary_state_cov, int complex_step=*) except 1
```

Comment:
Any particular reason you use 1 as the except val?  IIUC -1 is standard.

Reply:
No reason, simply that exceptions require a return value. I guess the Cython docs have except -1, but since the other SS code has return 1 I suppose that we can keep it as is?

Reply:
It looks like most of these are returning 0 unconditionally.  Getting rid of the return value/declaration would obviate the need to have the `except 1` clause at all.  The -1 seems to be a default elsewhere (namely pandas), but this is definitely not a big deal.

Reply:
Yes, that's definitely doable. I guess I vaguely thought maybe returning a fixed C value would be faster than the default (which is I guess a Python None object, maybe?), but I have no real reason to believe that.

Reply:
I think `cdef void foo` might achieve both goals.